src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,710 @@
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+        result = await get_category_item(category_name='integer', item_name='info')
+        assert result is None
+
+        result = await get_category_item(category_name='boolean', item_name='data')
+        assert result is None
+        await self.teardown_method()
+
+    async def test_get_category_all_items(self):
+        """
+        Test get_category_all_items method returns full "dictionary" of category_value
+        :assert:
+            1.  Values in dictionary are as expected
+            2. default doesn't get updated when value does
+        """
+        await  self.setup_method()
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+
+        result = await get_category_all_items(category_name='boolean')
+        assert result['info']['description'] == 'boolean type with default False'
+        assert result['info']['type'] == 'boolean'
+        assert result['info']['default'] == 'False'
+        assert result['info']['value'] == 'False'
+
+        await  set_category_item_value_entry(category_name='boolean', item_name='info',
+                                             new_value_entry='True')
+        result = await get_category_all_items(category_name='boolean')
+        assert result['info']['default'] == 'False'
+        assert result['info']['value'] == 'True'
+
+        await self.teardown_method()
+
+    async def test_get_category_all_items_empty(self):
+        """
+        Test get_category_all_items doesn't return anything if category_name doesn't exist
+        :assert:
+            Assert None gets returned when category_name does not exist
+        """
+        await self.setup_method()
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+
+        result = await get_category_all_items(category_name='integer')
+        assert result is None
+        await self.teardown_method()
+
+    async def test_get_all_category_names(self):
```

Comment:
duplicate test? sounds like its a test for " " should be trimmed and empty string should not be allowed (a bug?)? though `  `  spaces are valid password but only for password?